timon/plugins/http_server/views.py

```diff
@@ -94,3 +133,32 @@ async def force_probe_run(probename):
         "msg": probe.msg,
     }
     return data_to_return
+
+
+@app.route("/rescheduler/probes/", methods=['POST'])
+async def reschedule_probes():
+    """
+    Reschedule a specific probe to have
+    Params in request body:
+        - probenames
+        - timestamp (optional, default=time.time())
+    """
+    strdata = await request.get_data()
+    data = json.loads(strdata)
+    probenames = data["probenames"]
+    new_schedular = data.get("timestamp", time.time())
+    heap = app.tmoncfg.get_queue().heap
+    probes_to_reschedule = []
+    other_probes = []
+    for idx in range(len(heap)):
+        prb_schedular = heappop(heap)
+        if prb_schedular[1] in probenames:
+            probes_to_reschedule.append(prb_schedular)
+        else:
+            other_probes.append(prb_schedular)
+    for prb_info in other_probes:
+        heappush(heap, prb_info)
+    for prb_info in probes_to_reschedule:
+        prb_info[0] = new_schedular
+        heappush(heap, prb_info)
+    return "OK"
```

Comment:
doing N pops and N pushes is probably slower than
changing values, converting to list and recreating a heap

A also don't think you have any advantage by having three for loops 
just having one loop, doing a pop changing or not changing the time and pushing it should do the job, no?

Reply:
oops. no doesn't work with one loop now I get it

Reply:
back to iterating through the list (each heap should have an associated list object)
iterate through the list and create a new heap object at the end from the modified list
This is probably rather fast, as the list is 'almost' correct. SO the heap-sort doesn't have to do a lot (probably)

Reply:
Yes right, its slower
But I didn't want to change the heap object reference, that's why i removed all elements in it and repushed them ..

Reply:
to discuss if it's really needed

Reply:
yes let's discuss and change.

And in a future refactoriung effort we should abstract away the queue into a class so that changing the implementation won't impact the web views or other parts of the code.

Reply:
suggested implementation

```

new_sched_time = data.get("timestamp", time.time())
for entry in heap:
    if entry[1]  in probenames:
        entry[0] = new_sched_time
heapq.heapify(heap)
```